slides/_naming_for_clarity.qmd

```diff
@@ -1,43 +1,55 @@
-## Naming For Clarity {.smaller}
+<!-- ------------------------------------------------------------------------------ -->
+# Naming Clarity
+<!-- ------------------------------------------------------------------------------ -->
 
-It may seem inconsequential, but carefully naming variables and methods can improve the
-readability of code massively and can help to make code self-documenting.
+## Naming for clarity {.smaller}
 
-A few naming tips and conventions:
+\
 
-::: {.incremental}
-- The name should show the intention, think about how someone else might read it (this could be future you)
-- Use pronounceable names e.g.
-- ```
- ms    --> mass
- chclt --> chocolate
- stm   --> stem
- ```
-- avoid abbreviations and single letter variable names where possible
-- Use names that can be searched
-- One word per concept e.g. choose one of `put`, `insert`, `add` in the same code base
-:::
+It may seem inconsequential, but carefully naming variables and methods can greatly
+improve the readability of code.
+
+\
+
+Since _"code is read more than it is run"_ this is important for future you, but also
+for anyone you collaborate with or who might use your code in future.
+
+\
+
+It helps to make code self-documenting, reducing future bugs due to misunderstandings.
 
-## Naming For Clarity {.smaller}
+\
+
+Here we cover some key considerations when writing code.
+
+<!-- ------------------------------------------------------------------------------ -->
+
+## Naming for clarity {.smaller}
 
 ::: {.incremental}
-- Plurals to indicate groups, e.g. a list of dog objects would be `dogs`, not `dog_list`
-- Describe content rather than storage type e.g.
-- ```
-array       --> dogs
-age_int     --> age
-country_set --> countries
-```
-- Naming booleans, use prefixes like `is`, `has` or `can` and avoid negations like `not_green` e.g.
-- ```
-purple    --> is_purple
-not_plant --> is_plant
-sidekick  --> has_sidekick
-```
-- Keep it simple and use technical terms where appropriate
+- Show the intention -- how will someone else (future you) read it?
+- Use readable, pronounceable, memorable, and searchable names:
+  ```
+  ms    --> mass
+  chclt --> chocolate
+  stm   --> stem
+  ```
+  avoid abbreviations and single letters unless clear
```

Comment:
Maybe be more precise about "unless clear" -- e.g., "unless the name is commonly used in the wider research community" or so